From: Colin Ian King <colin.king@canonical.com>
Date: Thu, 17 Aug 2017 10:01:07 +0100
Subject: net: hns3: ensure media_type is unitialized
Patch-mainline: v4.14-rc1
Git-commit: fd07a62d58c16b1fdd46f4099791c5bd242908c3
References: bsc#1104353 FATE#326415

Media type is only set if h->ae_algo->ops->get_media_type is called
so there is a possibility that media_type is uninitialized when it is
used a switch statement.  Fix this by initializing media_type to
HNAE3_MEDIA_TYPE_UNKNOWN.

Detected by CoverityScan, CID#1452624("Uninitialized scalar variable")

Fixes: 496d03e960ae ("net: hns3: Add Ethtool support to HNS3 driver")
Signed-off-by: Colin Ian King <colin.king@canonical.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 drivers/net/ethernet/hisilicon/hns3/hns3pf/hns3_ethtool.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/ethernet/hisilicon/hns3/hns3pf/hns3_ethtool.c
+++ b/drivers/net/ethernet/hisilicon/hns3/hns3pf/hns3_ethtool.c
@@ -304,7 +304,7 @@ static int hns3_get_link_ksettings(struc
 	struct hnae3_handle *h = priv->ae_handle;
 	u32 supported_caps;
 	u32 advertised_caps;
-	u8 media_type;
+	u8 media_type = HNAE3_MEDIA_TYPE_UNKNOWN;
 	u8 link_stat;
 	u8 auto_neg;
 	u8 duplex;
